File: LEETCODE_CPP/392.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int m = s.size();
        int n = t.size();
        int i = 0, j = 0;
        
        while(i < m && j < n) {
            if(s[i] == t[j]) {
                i++;
            }
            j++;
        }

        if (i == m) {
            return 1;
        } else {
            return 0;
        }
    }
};

int main () {

    Solution p;

    vector<vector<string>> testCases = {
        {"abc","ahbgdc"},
        {"axc","ahbgdc"},
    };

    for (auto testCase : testCases) {
        cout << p.isSubsequence(testCase[0], testCase[1]) << endl; 
    }

    return 0;
}

File: LEETCODE_PYTHON/392.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        i = 0
        j = 0

        while i < len(s) and j < len(t):
            if (s[i] == t[j]):
                i += 1
            j += 1

        if i == len(s):
            return True
        else:
            return False 

def main():
    solution = Solution()
    testCases = [
        ["abc","ahbgdc"],
        ["axc","ahbgdc"],
    ]

    for testCase in testCases:
        result = solution.isSubsequence(testCase[0],testCase[1])
        print(result)

main()